File: base/src/main/java/com/tsy/base/tool/PreferencesLoader.java
package com.tsy.base.tool;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.annotation.StringRes;

/**
 * Created by jay on 17/7/31.
 */

public class PreferencesLoader {

    private SharedPreferences sharedPreferences;
    private Context context;

    public PreferencesLoader(Context context) {
        this.context = context;
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public void saveString(@StringRes int keyResId, String value) {
        String key = context.getString(keyResId);
        saveString(key, value);
    }

    public void saveString(String key, String value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public String getString(String key) {
        return sharedPreferences.getString(key, "");
    }

    public void saveLong(@StringRes int keyResId, long value) {
        String key = context.getString(keyResId);
        saveLong(key, value);
    }

    public void saveLong(String key, long value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putLong(key, value);
        editor.apply();
    }

    public long getLong(String key) {
        return sharedPreferences.getLong(key, 0L);
    }
}

File: base/src/main/java/com/tsy/base/tool/PreferencesLoader.kt
package com.tsy.base.tool

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.support.annotation.StringRes

/**
 * Created by jay on 17/7/31.
 */

class PreferencesLoader(private val context: Context) {

    private val sharedPreferences: SharedPreferences

    init {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
    }

    fun saveString(@StringRes keyResId: Int, value: String) {
        val key = context.getString(keyResId)
        saveString(key, value)
    }

    fun saveString(key: String, value: String) {
        val editor = sharedPreferences.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun getString(key: String): String {
        return sharedPreferences.getString(key, "")
    }

    fun saveLong(@StringRes keyResId: Int, value: Long) {
        val key = context.getString(keyResId)
        saveLong(key, value)
    }

    fun saveLong(key: String, value: Long) {
        val editor = sharedPreferences.edit()
        editor.putLong(key, value)
        editor.apply()
    }

    fun getLong(key: String): Long {
        return sharedPreferences.getLong(key, 0L)
    }
}
